### PR TITLE
Refactor/use error handling middleware

### PR DESCRIPTION
**Pull request by MatthewEllisTx**
# Description

used .catch(next) to handle erorrs

## Why was this work done?

Got tired of copying/pasting, so I just changed it all to something shorter

### CODE REVIEW

File: api/provider/providerRouter.js [hunk 1]
```diff
@@ -173,7 +170,7 @@ router.get('/', function (req, res) {
  *        description: 'Profile not found'
  */
 
-router.get('/getServiceProviders', function (req, res) {
+router.get('/getServiceProviders', (req, res) => {
```
Comment (DomenicScarcella) [inline api/provider/providerRouter.js:173]: Why no "next" on Line 173, and no .catch(next) at the end of the code block?  This seems to be the only changed instance in all the files that doesn't follow that pattern.

File: api/app.js [hunk 2]
```diff
@@ -69,31 +68,11 @@ app.use(['/api/household', '/api/households'], householdRouter);
...
-  if (res.locals.status) {
-    res.status(res.locals.status || 500);
-    const errObject = { error: res.locals.error, message: res.locals.message };
-    return res.json(errObject);
-  }
-  next(err);
+app.use((err, req, res, next) => { // eslint-disable-line
+  res.status(err.status || 500).json({
+    message: err.message,
+    stack: err.stack,
```
Comment (jaypdl) [inline api/app.js:74]: I love me some error handling!

In this case, showing the 'stack' in a production site can open it up to vulnerabilities.
Consider adding a check for if it's in production based on the env. 

Although I realize the frontend team is using a deployed backend that Heroku marks as production for their testing. So actually, maybe just leave a comment for the next cohort.

File: api/provider/providerRouter.js (hunk 1 above)
Comment (jaypdl) [reply api/provider/providerRouter.js:173]: Good catch @DomenicScarcella

**Review by jaypdl — CHANGES_REQUESTED** [review]
I think just change what @DomenicScarcella mentioned.

Great job adding the error handling!

File: api/provider/providerRouter.js (hunk 1 above)
Comment (MatthewEllisTx) [reply api/provider/providerRouter.js:173]: This one didn't have a .catch( err => ... ) so I didn't add it. I can change it though

File: api/app.js (hunk 2 above)
Comment (MatthewEllisTx) [reply api/app.js:74]: Good catch. I wasn't sure if either the stack or message would be a problem, or both

**Pull request merged**